src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,266 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Alexander Surkov (alex@sourcerer.io)
+
+package app
+
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ObjectLoader
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.text.SimpleDateFormat
+import java.util.Date
+
+/**
+ * Represents a code line in a file revision.
+ */
+class RevCommitLine(val commit: RevCommit, val file: String, val line: Int)
+
+/**
+ * Represents a code line in repo's history.
+ *
+ * TODO(Alex): the text arg is solely for testing proposes (remove it)
+ */
+class CodeLine(val from: RevCommitLine, val to: RevCommitLine, val text: String) {
+
+    // TODO(alex): oldId and newId may be computed as a hash built from commit,
+    // file name and line number, if we are going to send the data outside a
+    // local machine.
+
+    /**
+     * Id of the code line in a revision when the line was added. Used to
+     * identify a line and update its lifetime computed at the previous
+     * iteration.
+     */
+    val oldId: String = ""
```

Comment:
yeah, we should avoid sending anything to server except for stuff we can't do without. we don't need some blogger saying to everyone that sourcerer uploads per-line information of your proprietary code. if it can be done locally, let's do it locally.